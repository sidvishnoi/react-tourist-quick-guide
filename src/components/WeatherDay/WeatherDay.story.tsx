import {
  boolean,
  number,
  select,
  text,
  withKnobs,
} from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import * as React from 'react';
import WeatherDay, { WeatherDayProps } from '.';
import weatherCodes from './weather-codes';

const stories = storiesOf('WeatherDay', module).addDecorator(withKnobs);

stories.add('weather', () => {
  const condition = select(
    'weather condition',
    Array.from(weatherCodes.keys()),
    'Sunny',
  );
  const props: WeatherDayProps = {
    icon: weatherCodes.get(condition),
    summary: condition,
    temperature: number('temperature', 30, {
      max: 100,
      min: -50,
      range: true,
      step: 1,
    }),
  };

  const isSmall = boolean('isSmall', false);

  return (
    <div
      style={{
        border: '1px solid #000',
        maxWidth: isSmall ? '100px' : '300px',
      }}
    >
      <WeatherDay
        isSmall={isSmall}
        temperature={props.temperature}
        icon={props.icon}
        summary={props.summary}
      />
    </div>
  );
});
