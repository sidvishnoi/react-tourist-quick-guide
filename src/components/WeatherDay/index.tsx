import * as React from 'react';
import styled from 'styled-components';

export interface WeatherDayProps {
  temperature: number;
  icon: number;
  summary: string;
}

const WeatherDay = styled.div<{ isSmall: boolean }>`
  --padding: ${props => (props.isSmall ? '1rem' : '2rem')};
  background-repeat: no-repeat;
  background-position: top right;
  background-size: ${props => (props.isSmall ? '2rem' : 'auto')};
  padding: var(--padding);
  transition: all 0.2s ease;
  flex-grow: 1;

  span {
    font-size: ${props => (props.isSmall ? '1.2rem' : '2.5rem')};
  }

  p {
    font-size: 70%;
    font-style: italic;
    margin: 0;
    max-width: 70%;
  }

  @media print {
    background-size: 0;
    --padding: 0 0.5rem;

    span {
      font-size: ${props => (props.isSmall ? '1.2rem' : '2rem')};
    }
  }
`;

export default function(props: WeatherDayProps & { isSmall?: boolean }) {
  const { temperature, icon, summary, isSmall = false } = props;
  const backgroundImage = `url(/images/weather-icons/${icon}.png)`;
  return (
    <WeatherDay
      isSmall={isSmall}
      style={{ backgroundImage }}
      className="weather-day"
    >
      <span>{temperature}</span>Â°C
      <p>{summary}</p>
    </WeatherDay>
  );
}
