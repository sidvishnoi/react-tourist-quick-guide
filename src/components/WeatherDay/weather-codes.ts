/* istanbul ignore file */

export default new Map([
  ['Sunny', 113],
  ['Partly cloudy', 116],
  ['Cloudy', 119],
  ['Overcast', 122],
  ['Mist', 143],
  ['Patchy rain possible', 176],
  ['Patchy snow possible', 179],
  ['Patchy sleet possible', 182],
  ['Patchy freezing drizzle possible', 185],
  ['Thundery outbreaks possible', 200],
  ['Blowing snow', 227],
  ['Blizzard', 230],
  ['Fog', 248],
  ['Freezing fog', 260],
  ['Patchy light drizzle', 263],
  ['Light drizzle', 266],
  ['Freezing drizzle', 281],
  ['Heavy freezing drizzle', 284],
  ['Patchy light rain', 293],
  ['Light rain', 296],
  ['Moderate rain at times', 299],
  ['Moderate rain', 302],
  ['Heavy rain at times', 305],
  ['Heavy rain', 308],
  ['Light freezing rain', 311],
  ['Moderate or heavy freezing rain', 314],
  ['Light sleet', 317],
  ['Moderate or heavy sleet', 320],
  ['Patchy light snow', 323],
  ['Light snow', 326],
  ['Patchy moderate snow', 329],
  ['Moderate snow', 332],
  ['Patchy heavy snow', 335],
  ['Heavy snow', 338],
  ['Ice pellets', 350],
  ['Light rain shower', 353],
  ['Moderate or heavy rain shower', 356],
  ['Torrential rain shower', 359],
  ['Light sleet showers', 362],
  ['Moderate or heavy sleet showers', 365],
  ['Light snow showers', 368],
  ['Moderate or heavy snow showers', 371],
  ['Light showers of ice pellets', 374],
  ['Moderate or heavy showers of ice pellets', 377],
  ['Patchy light rain with thunder', 386],
  ['Moderate or heavy rain with thunder', 389],
  ['Patchy light snow with thunder', 392],
  ['Moderate or heavy snow with thunder', 395],
]);
