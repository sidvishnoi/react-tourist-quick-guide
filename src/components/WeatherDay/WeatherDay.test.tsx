import { mount } from 'enzyme';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
Enzyme.configure({ adapter: new Adapter() });

import WeatherDay, { WeatherDayProps } from '.';

const props: WeatherDayProps = {
  icon: 'sun',
  isSmall: false,
  summary: 'sunny',
  temperature: 30,
  unit: 'F',
};

describe('Components - WeatherDay', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<WeatherDay {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should show temperature value and summary with correct unit', () => {
    const output = mount(<WeatherDay {...props} />);

    expect(output.find('span').text()).toEqual(`${props.temperature}`);
    expect(output.find('p').text()).toEqual(props.summary);
  });
});
