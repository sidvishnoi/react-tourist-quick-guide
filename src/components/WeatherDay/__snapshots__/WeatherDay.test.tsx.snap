// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - WeatherDay matches snapshot 1`] = `
<div
  className="weather-day sc-bdVaJa kuIetW"
  style={
    Object {
      "backgroundImage": "url(/images/weather-icons/113.png)",
    }
  }
>
  <span>
    30
  </span>
  Â°C
  <p>
    sunny
  </p>
</div>
`;
